mobile/native/src/db/custom_types.rs

```diff
@@ -168,6 +170,54 @@ impl FromSql<Text, Sqlite> for PositionState {
     }
 }
 
+impl ToSql<Text, Sqlite> for HtlcStatus {
+    fn to_sql(&self, out: &mut Output<Sqlite>) -> serialize::Result {
+        let text = match *self {
+            HtlcStatus::Pending => "Pending",
+            HtlcStatus::Succeeded => "Succeeded",
+            HtlcStatus::Failed => "Failed",
+        };
+        out.set_value(text);
+        Ok(IsNull::No)
+    }
+}
+
+impl FromSql<Text, Sqlite> for HtlcStatus {
+    fn from_sql(bytes: backend::RawValue<Sqlite>) -> deserialize::Result<Self> {
+        let string = <String as FromSql<Text, Sqlite>>::from_sql(bytes)?;
+
+        return match string.as_str() {
+            "Pending" => Ok(HtlcStatus::Pending),
+            "Succeeded" => Ok(HtlcStatus::Succeeded),
+            "Failed" => Ok(HtlcStatus::Failed),
+            _ => Err("Unrecognized enum variant".into()),
+        };
+    }
+}
+
+impl ToSql<Text, Sqlite> for Flow {
+    fn to_sql(&self, out: &mut Output<Sqlite>) -> serialize::Result {
+        let text = match *self {
+            Flow::Inbound => "Inbound",
+            Flow::Outbound => "Outbound",
+        };
+        out.set_value(text);
+        Ok(IsNull::No)
+    }
+}
+
+impl FromSql<Text, Sqlite> for Flow {
+    fn from_sql(bytes: backend::RawValue<Sqlite>) -> deserialize::Result<Self> {
+        let string = <String as FromSql<Text, Sqlite>>::from_sql(bytes)?;
+
+        return match string.as_str() {
+            "Inbound" => Ok(Flow::Inbound),
+            "Outbound" => Ok(Flow::Outbound),
+            _ => Err("Unrecognized enum variant".into()),
+        };
+    }
+}
+
```

Comment:
(unrelated to this particular change, just a meta observation) : it's sad that we need to do that manually with diesel :( can't `serde` deal with this?

Reply:
This boilerplate is annoying, I agree.